src/hyperlight_component_util/src/component.rs

```diff
@@ -112,8 +113,23 @@ pub fn read_component_single_exported_type<'a>(
                         Err(_) => panic!("invalid export section"),
                         Ok(ce) => {
                             if ce.kind == ComponentExternalKind::Type {
-                                last_idx = Some(ctx.types.len());
                                 ctx.types.push(raw_type_export_type(&ctx, &ce).clone());
+
+                                // picks the world index if world is passed in the proc_macro
+                                // else picks the index of last type, exported by core module
+                                if world_name.is_some() {
+                                    let world = world_name.clone().unwrap();
+                                    match ce.name {
+                                        wasmparser::ComponentExportName(name) => {
+                                            if name.eq_ignore_ascii_case(&world) {
+                                                picks_one_world_or_last_idx =
+                                                    Some(ctx.types.len() - 1);
```

Comment:
Why is this minus one here? Previous it was just `last_idx = Some(ctx.types.len());`